__Задача 1__

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X)`, `D(X)`.

5 вопросов - это количество испытаний. Обозначим n = 5
В каждом вопросе 4 варианта ответа, один из которых правильный. Следовательно вероятность угадать правильный ответ в каждом вопросе равна p = 1/4.
X - количество правильно угаданных ответов. X может принять значение от 0 до 5.
Рассчитаем вероятность появления события для каждого испытания:
$$P(X = 0) = C_n^k p^k (1 - p)^{n - k} = C_5^0 \cdot 0.25^0 \cdot 0.75^5 = \frac{5!}{0! \: (5-0)!} \cdot 0.25^0 \cdot 0.75^5 = 1 \cdot 1 \cdot 0.237 = 0.237$$

$$P(X = 1) =  C_5^1 \cdot 0.25^1 \cdot 0.75^4 = \frac{5!}{1! \: (5-1)!} \cdot 0.25^1 \cdot 0.75^4 = 1 \cdot 0.25 \cdot 0.316 = 0.396$$

Дальше я решил не считать все вручную, потому что долго набирать. Поэтому написал функцию которая возвращает вероятность в каждом испытании

In [2]:
import numpy as np

In [36]:
def p_bernoulli(n:int,p:float):
    res = []
    for k in range(0,n+1): 
        c = np.math.factorial(n)/(np.math.factorial(k)* np.math.factorial(n-k))
        p_x = c * pow(p,k) * pow((1-p),n-k)
        res.append(round(p_x,3))
    return res
print(p_bernoulli(5,0.25))

[0.237, 0.396, 0.264, 0.088, 0.015, 0.001]


Получаем:
$$P(X = 0) = 0.237$$ 
$$P(X = 1) = 0.396$$ 
$$P(X = 2) = 0.264$$ 
$$P(X = 3) = 0.088$$ 
$$P(X = 4) = 0.015$$ 
$$P(X = 5) = 0.001$$ 

Мат. ожидание $$M(X) = np = 5 \cdot 0.25 = 1.25$$

Дисперсия $$D(X) = np(1 - p) = 5 \cdot 0.25 \cdot (1 - 0.25) = 0.9375$$

__Задача 2__

Пользователь получает в среднем 10 писем со спамом на свой почтовый ящик за сутки. Найти число `N`, такое, что с вероятностью 0.95 пользователь получит не более `N` писем со спамом за текущий день.

Здесь у нас Пуассоновское распределение с лямбда = 10. Для того чтобы найти такое число писем N, мы должны сложить вероятности событий получения 0,1,2 ... N писем до того момента пока вероятность не будет равна 0,95.

In [15]:
lambda_ = 10
k = 0
p_sum = 0
while True:
    p = (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)
    p_sum +=p
    if p_sum <= 0.95:
        k += 1
    else:
        print(f'p = {p_sum}')
        print(f'N = {k}')
        break

p = 0.9512595966960214
N = 15


Получаем что пользователь получит не более 15 писем в день с вероятностью p = 0.9512595966960214.

__Задача 3__

Производятся выстрелы по мишени. Вероятность попадания в мишень при одном выстреле равна 0.01. Сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание?

_Подсказка_: 1) "Вероятность попасть `k` раз при `n` выстрелах" - на какое распределение это похоже? 2) А если нам нужна вероятность `P(X >= k)`, а не `P(X = k)`? 3) Здесь предстоит немножко покодить.

Здсь налицо Биномиальное распределение. Чтобы найти итоговое количество выстрелов нам нужен цикл, который будет высчитывать вероятность 6 попаданий при каждом последующим выстреле и сумма этих вероятностей не должна быть меньше 0.9.Начинать цикл будем при n = 6 - это минимальное число выстрелов при которых возможно наступление события.

In [23]:
def p_bernoulli(n:int,p:float,k:int):
    c = np.math.factorial(n)/(np.math.factorial(k)* np.math.factorial(n-k))
    p_x = c * pow(p,k) * pow((1-p),n-k)
    return p_x

n = 6
p_sum
while True:
    p_sum += p_bernoulli(n,0.01,6)
    if p_sum <= 0.9:
        n += 1
    else:
        print(p_sum)
        print(n)
        break



0.9185129699543314
230


Соответсвенно вероятностью 0.9185129699543314 стрелок попадет в мишень 6 раз при 230 выстрелах.